src/main/java/me/mrCookieSlime/Slimefun/Events/AndroidMineEvent.java

```diff
@@ -0,0 +1,59 @@
+package me.mrCookieSlime.Slimefun.Events;
+
+import me.mrCookieSlime.Slimefun.androids.ProgrammableAndroid;
+import org.bukkit.block.Block;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+
+/**
+ * This event is fired before a miner android mines a block.
+ * If this event is cancelled, the block will not be mined.
+ */
+public class AndroidMineEvent extends Event implements Cancellable {
+
+    private static final HandlerList handlers = new HandlerList();
+
+    private final Block block;
+    private final ProgrammableAndroid android;
+    private boolean cancelled;
+
+    /**
+     * @param block - mined block
+     * @param android - android
+     */
+    public AndroidMineEvent(Block block, ProgrammableAndroid android) {
+        this.block = block;
+        this.android = android;
+    }
+
+    public static HandlerList getHandlerList() {
+        return handlers;
+    }
+
+    public HandlerList getHandlers() {
+        return handlers;
+    }
+
+    /**
+     * @return the mined block
+     */
+    public Block getBlock() {
+        return this.block;
+    }
+
+    /**
+     * @return the android
+     */
+    public ProgrammableAndroid getAndroid() { return this.android;}
```

Comment:
First off, please format this like the other methods.

Second, this would only return the instance of SlimefunItem, you should better return the Block of the android instead to correctly identify the android.

Reply:
> Second, this would only return the instance of SlimefunItem, you should better return the Block of the android instead to correctly identify the android.

So if I understand correctly, it should return the Block b parameter of the ProgrammableAndroid#mine function?

Reply:
Correct